src/core/init/initial_seek_and_play.ts

```diff
@@ -63,7 +63,10 @@ function canPlay(
       if (seeking || stalled !== null) {
         return false;
       }
-      if (!shouldWaitForDataBeforeLoaded(isDirectfile)) {
+      if (!shouldWaitForDataBeforeLoaded(
+        isDirectfile,
+        mediaElement.hasAttribute("playsinline"))
```

Comment:
`playsinline` seems to be webkit-specific, isn't it?

If that's the case, this check should also be moved to compat, like most things that are both specific to few browsers and which are not specified by WHATWG or W3C.